lib/plugins/aws/package/compile/events/apiGateway/lib/usagePlan.test.js

```diff
@@ -288,3 +289,113 @@ describe('#compileUsagePlan()', () => {
     });
   });
 });
+
+describe('UsagePlan', () => {
+  const burstLimit = 98;
+  const rateLimit = 99;
+
+  const limit = 101;
+  const offset = 2;
+  const period = 'MONTH';
+
+  const quota = {
+    limit,
+    offset,
+    period,
+  };
+
+  const throttle = {
+    burstLimit,
+    rateLimit,
+  };
+
+  const fn = {
+    foo: {
+      events: [
+        {
+          http: {
+            integration: 'AWS',
+            request: { template: { 'application/x-www-form-urlencoded': null } },
+          },
+        },
+      ],
+    },
+  };
+
+  it('Should have values for throttle', () =>
+    runServerless({
+      fixture: 'usagePlan',
+      configExt: {
+        provider: {
+          name: 'aws',
+          usagePlan: {
+            throttle,
+          },
+        },
+        functions: fn,
+      },
+      cliArgs: ['package'],
+    }).then(({ cfTemplate }) => {
+      const cfResources = cfTemplate.Resources;
+
+      expect(cfResources.ApiGatewayUsagePlan.Properties.Throttle).to.not.be.undefined;
+
+      expect(cfResources.ApiGatewayUsagePlan.Properties.Throttle.BurstLimit).to.be.equal(
+        burstLimit
+      );
+      expect(cfResources.ApiGatewayUsagePlan.Properties.Throttle.RateLimit).to.be.equal(rateLimit);
+    }));
+
+  it('Should have values for quota', () =>
+    runServerless({
```

Comment:
Let's use single `runServerless` run to test all cases (we may create three `http` events).

Those runs are expensive, therefore we should keep it's count as minimal as possible

Reply:
Hmm, is that possible? `provider` is on level 1 along with `functions` but events are inside `functions->foo->events`. So there cannot be three events with different provider options. Am I missing something?

```
{
    functions: {
      foo: {
        events: [
          {
            http: {},
          },
          {
            http: {},
          }
        ],
      },
    },
    provider: {
      usagePlan: {},
    },
  }
```

Reply:
I think a single test for quota & throttle together should do the trick

Reply:
@andreizet indeed if we need to test different `provider` configurations we need different `runServerless` runs. It's totally justifiable in such case